app/components/emoji_picker/emoji_picker_row.js

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {
+    StyleSheet,
+    TouchableOpacity,
+    View
+} from 'react-native';
+
+import Emoji from 'app/components/emoji';
+
+export default class EmojiPickerRow extends Component {
+    static propTypes = {
+        emojiGutter: PropTypes.number.isRequired,
+        emojiSize: PropTypes.number.isRequired,
+        items: PropTypes.array.isRequired,
+        onEmojiPress: PropTypes.func.isRequired
+    }
+
+    shouldComponentUpdate(nextProps) {
+        return this.props.items.length !== nextProps.items.length;
+    }
+
+    renderEmojis = (emoji, index, emojis) => {
+        const {emojiGutter, emojiSize} = this.props;
+
+        const style = [
+            styles.emoji,
+            {
+                width: emojiSize,
+                height: emojiSize,
+                marginHorizontal: emojiGutter
+            }
+        ];
+        if (index === 0) {
+            style.push(styles.emojiLeft);
+        } else if (index === emojis.length - 1) {
+            style.push(styles.emojiRight);
+        }
+
+        if (!emoji) {
+            return (
+                <View
+                    key={index}
+                    style={style}
+                />
+            );
+        }
+
+        return (
+            <TouchableOpacity
+                key={emoji.name}
+                style={style}
+                onPress={() => {
+                    this.props.onEmojiPress(emoji.name);
+                }}
+            >
+                <Emoji
+                    emojiName={emoji.name}
+                    size={emojiSize}
+                />
+            </TouchableOpacity>
+        );
+    }
+
+    render() {
+        const {emojiGutter, items} = this.props;
+
+        return (
+            <View style={[styles.columnStyle, {marginVertical: emojiGutter}]}>
+                {items.map(this.renderEmojis)}
+            </View>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    columnStyle: {
```

Comment:
if the row is not completely filled then this will be rendered in a very weird way

Reply:
This line here actually takes care of that https://github.com/mattermost/mattermost-mobile/pull/1136/files#diff-e7ebfda4e52a10cd0b0b459e39d61bb6R99

I think what you saw in your DM was maybe an instance of a custom emoji not loading.